jobs/build-snaps/build-release-eks-snaps.groovy

```diff
@@ -0,0 +1,154 @@
+
+                    for snap in ${CK_SNAPS}
+                    do
+                        CK_SNAP_REPO="\${CK_SNAP_REPO_PREFIX}\${snap}"
+                        EKS_SNAP="\${snap}${EKS_SUFFIX}"
+
+                        if git ls-remote --exit-code --heads \${CK_SNAP_REPO} \${CK_SNAP_BRANCH}
+                        then
+                            echo "Getting \${snap} from \${CK_SNAP_BRANCH} branch."
+                            git clone \${CK_SNAP_REPO} --branch \${CK_SNAP_BRANCH} --depth 1 \${EKS_SNAP}
+                        else
+                            echo "ERROR: \${CK_SNAP_BRANCH} branch not found in the \${CK_SNAP_REPO} repo."
+                            exit 1
+                        fi
+
+                        # eks snaps have a suffix and different base than ck; adjust snapcraft.yaml
+                        cd \${EKS_SNAP}
+                        sed -i -e "s/^name: \${snap}/name: \${EKS_SNAP}/" \
+                               -e "s/^base: .*/base: ${EKS_BASE}/" snapcraft.yaml
+                        cd -
+                    done
+                """
+            }
+        }
+        stage('Setup Build Container'){
+            steps {
+                /* override sh for this step:
+
+                 Needed because cilib.sh has some non POSIX bits.
+                 */
+                sh """#!/usr/bin/env bash
+                    . \${WORKSPACE}/cilib.sh
+
+                    ci_lxc_launch ubuntu:20.04 ${lxc_name}
+                    until sudo lxc shell ${lxc_name} -- bash -c "snap install snapcraft --classic"; do
+                        echo 'retrying snapcraft install in 3s...'
+                        sleep 3
+                    done
+
+                    for snap in ${CK_SNAPS}
+                    do
+                        EKS_SNAP="\${snap}${EKS_SUFFIX}"
+
+                        echo "Copying \${EKS_SNAP} into container."
+                        sudo lxc file push \${EKS_SNAP} ${lxc_name}/ -p -r
+                    done
+                """
+            }
+        }
+        stage('Build Snaps'){
+            steps {
+                sh """
+                    for snap in ${CK_SNAPS}
+                    do
+                        EKS_SNAP="\${snap}${EKS_SUFFIX}"
+
+                        echo "Building \${EKS_SNAP}."
+                        # NB: kubelet snapcraft.yaml defines an alias, but
+                        # aliases wont work in 'bash -c'. Define a func that
+                        # is functionally equivalent to the alias. Single quote
+                        # it because we need literal vars passed to 'bash -c'.
+                        sudo lxc shell ${lxc_name} -- bash -c \
+                            'set -a; add-arg() { \$SNAPCRAFT_PART_BUILD/shared/add-arg-to-configure-hook \$@; }; set +a; '"cd /\${EKS_SNAP};"' SNAPCRAFT_BUILD_ENVIRONMENT=host snapcraft'
+                    done
+                """
+            }
```

Comment:
this is begging for a parallel step, but all in the same lxd could be tricky if the builds don't like that.

Reply:
Yeah, it super stinks because each snap dupes the k8s repo clone, e.g. https://git.launchpad.net/snap-kubelet/tree/snapcraft.yaml.in#n59

I added a backlog story (CK-369) to noodle on improvements.